### PR TITLE
Introduce Kind extension that supports autoloading

### LINKED ISSUE #23020: Add support for kind
### Description

Kind is a tool for creating locally dev clusters for kubernetes.
So, it would be nice if we had the same conventions we use for minikube applied to kind.
Additionally, kind does require a manual step to load images to the cluster and this `breaks` the kubernetes deployment experience. 

### Implementation ideas

We need an extension that tunes manifests as minikube does (so we could possibly share code here).

We also need to invoke `kind load docker image xxxx` after each docker, jib, buildpack build that uses the docker daemon.

### PR DESCRIPTION
**Pull request by iocanel**
Resolves: #23020

### CODE REVIEW

File: extensions/container-image/spi/src/main/java/io/quarkus/container/spi/ContainerImageBuilderBuildItem.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+
+package io.quarkus.container.spi;
+
+import io.quarkus.builder.item.MultiBuildItem;
+
+public final class ContainerImageBuilderBuildItem extends MultiBuildItem {
```
Comment (geoand) [inline extensions/container-image/spi/src/main/java/io/quarkus/container/spi/ContainerImageBuilderBuildItem.java:6]: Can you add some Javadoc on what the purpose of this build item is?

Comment (geoand) [reply extensions/container-image/spi/src/main/java/io/quarkus/container/spi/ContainerImageBuilderBuildItem.java:6]: Bump this

**Review by geoand — APPROVED** [review]

**Comment by gastaldi** [comment]
Nice, however, should this extension live in core? I mean, we have already https://github.com/quarkiverse/quarkus-helm.

It just feels like being a Quarkiverse extension to me 😃  WDYT?

**Comment by geoand** [comment]
Well, it's similar to our Minikube support - making it super easy for users to work with popular developer-oriented k8s "distributions".

**Comment by geoand** [comment]
@iocanel mind rebasing onto main just to avoid any nasty surprises (as the PR is a few days old)

**Comment by gsmet** [comment]
I did the rebase as there were conflicts but I was unable to push to your branch so I created another PR: https://github.com/quarkusio/quarkus/pull/24449 .

**Pull request closed without merging**